subsys/net/lib/lwm2m/lwm2m_rw_oma_tlv.c

```diff
@@ -935,23 +935,53 @@ static int do_write_op_tlv_item(struct lwm2m_message *msg)
 	return ret;
 }
 
+static int write_tlv_resource(struct lwm2m_message *msg, struct oma_tlv *tlv)
+{
+	int ret;
+
+	if (msg->in.block_ctx) {
+		msg->in.block_ctx->res_id = tlv->id;
+	}
+
+	msg->path.res_id = tlv->id;
+	msg->path.level = 3U;
+	ret = do_write_op_tlv_item(msg);
+
+	/*
+	 * ignore errors for CREATE op
+	 * for OP_CREATE and BOOTSTRAP WRITE: errors on
+	 * optional resources are ignored (ENOTSUP)
+	 */
+	if (ret < 0 &&
+	    !((ret == -ENOTSUP) &&
+	      (msg->ctx->bootstrap_mode ||
+	       msg->operation == LWM2M_OP_CREATE))) {
+		return ret;
+	}
+
+	return 0;
+}
+
 int do_write_op_tlv(struct lwm2m_message *msg)
 {
 	struct lwm2m_engine_obj_inst *obj_inst = NULL;
 	size_t len;
 	struct oma_tlv tlv;
 	int ret;
 
-	/* In case of Firmware object Package resource go directly to the
+	/* In case of block transfer go directly to the
 	 * message processing - consecutive blocks will not carry the TLV
 	 * header.
 	 */
-	if (msg->in.block_ctx != NULL && msg->in.block_ctx->ctx.current > 0 &&
-	    msg->path.obj_id == 5 && msg->path.res_id == 0) {
+	if (msg->in.block_ctx != NULL && msg->in.block_ctx->ctx.current > 0) {
+		msg->path.res_id = msg->in.block_ctx->res_id;
+		msg->path.level = 3U;
 		ret = do_write_op_tlv_item(msg);
 		if (ret < 0) {
 			return ret;
 		}
+
+		return 0;
```

Comment:
This "return 0;" seems to cause a problem.
If the current packet contains writes to other resources in the object, those will not be processed. Only the first item is written, and function returns without processing the rest.
If I remove the "return 0", it seems to resolve my problem and the other writes are processed. However, that "return 0" was specifically added for a reason, so I am not confident that is the correct fix.